http-clients/src/test/java/feign/TextDelegateDecoderTest.java

```diff
@@ -0,0 +1,113 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+
+
+public final class TextDelegateDecoderTest {
+    private static final String DELEGATE_RESPONSE = "delegate response";
+
+    @ClassRule
+    public static final DropwizardAppRule<Configuration> APP = new DropwizardAppRule<>(TestServer.class,
+            "src/test/resources/test-server.yml");
+
+    private TestServer.TestService service;
+    private Map<String, Collection<String>> headers;
+    private Decoder delegate;
+    private Decoder textDelegateDecoder;
+
+    @Before
+    public void before() {
+        delegate = mock(Decoder.class);
+        headers = Maps.newHashMap();
+        textDelegateDecoder = new TextDelegateDecoder(delegate);
+
+        String endpointUri = "http://localhost:" + APP.getLocalPort();
+        service = FeignClients.standard().createProxy(Optional.<SSLSocketFactory>absent(), endpointUri,
+                TestServer.TestService.class);
+    }
+
+    @Test
+    public void testUsesStringDecoderWithTextPlain() throws Exception {
+        headers.put(HttpHeaders.CONTENT_TYPE, ImmutableSet.of(MediaType.TEXT_PLAIN));
+        Response response = Response.create(200, "OK", headers, "text response", Charsets.UTF_8);
+        Object decodedObject = textDelegateDecoder.decode(response, String.class);
+
+        assertEquals(decodedObject, "text response");
+        verifyZeroInteractions(delegate);
+    }
+
+    @Test
+    public void testUsesDelegateWithNoHeader() throws Exception {
+        when(delegate.decode((Response) any(), (Type) any())).thenReturn(DELEGATE_RESPONSE);
+        Response response = Response.create(200, "OK", headers, new byte[0]);
+        Object decodedObject = textDelegateDecoder.decode(response, String.class);
+
+        assertEquals(decodedObject, DELEGATE_RESPONSE);
+    }
+
+    @Test
+    public void testUsesDelegateWithComplexHeader() throws Exception {
+        headers.put(HttpHeaders.CONTENT_TYPE, ImmutableSet.of(MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON));
+        when(delegate.decode((Response) any(), (Type) any())).thenReturn(DELEGATE_RESPONSE);
+        Response response = Response.create(200, "OK", headers, new byte[0]);
+        Object decodedObject = textDelegateDecoder.decode(response, String.class);
+
+        assertEquals(decodedObject, DELEGATE_RESPONSE);
+    }
+
+    @Test
+    public void testUsesDelegateWithNonTextContentType() throws Exception {
+        headers.put(HttpHeaders.CONTENT_TYPE, ImmutableSet.of(MediaType.APPLICATION_JSON));
+        when(delegate.decode((Response) any(), (Type) any())).thenReturn(DELEGATE_RESPONSE);
+        Response response = Response.create(200, "OK", headers, new byte[0]);
+        Object decodedObject = textDelegateDecoder.decode(response, String.class);
+
+        assertEquals(decodedObject, DELEGATE_RESPONSE);
+    }
+
+    @Test
+    public void testStandardClientsUseTextDelegateEncoder() {
+        assertThat(service.getString("string"), is("string"));
+
+    }
```

Comment:
can we also cover that a null response comes back with content type text/plain and you get the right `Optional` behavior?